operator/internal/handlers/internal/gateway/tenant_secrets.go

```diff
@@ -61,3 +60,27 @@ func GetTenantSecrets(
 
 	return tenantSecrets, nil
 }
+
+// ExtractGatewaySecret reads a k8s secret into a manifest tenant secret struct if valid.
+func extractSecret(s *corev1.Secret, tenantName string) (*manifests.TenantSecrets, error) {
+	// Extract and validate mandatory fields
+	clientID, ok := s.Data["clientID"]
+	if !ok {
```

Comment:
Can these fields be "empty"? If not, should we add a check for emptyness?

```suggestion
	clientID := s.Data["clientID"]
	if clientID == "" {
```

Same for the other two options.

There are a lot of checks like this in the "extract storage secrets" code as well... If we remember we could also do this as one change in a subsequent PR.

Reply:
Let's do this here, it's not much work.